File: deaf_grandma.py
def deaf_grandma():
    goodbye = 0
    response = input("HEY KID!")
    while goodbye != 2:
        if response == "GOODBYE!" and goodbye == 0:
            response = input("LEAVING SO SOON?")
            goodbye = 1
        if(response == "GOODBYE!" and goodbye == 1):
            print("LATER, SKATER!")
            return
        if response.isupper() == True:
            response = input("NO, NOT SINCE 1946!")
        if response == "":
            response = input('WHAT?!')
        if response != "" and response.isupper() == False:
            response = input('SPEAK UP, KID!')
        

deaf_grandma()

File: deaf_grandma.js
function deafGrandma(count = 0) {
    let input = prompt('HEY KID!')
    if(input === ''){
        alert('WHAT?!');
    } else if(input === "GOODBYE!" && count === 0){
        alert('LEAVING SO SOON?')
        return deafGrandma(1);
    } else if(input === "GOODBYE!" && count === 1){
        alert('LATER, SKATER!')
        return;
    } else if(input === input.toUpperCase()){
        alert('NO, NOT SINCE 1946!');
    } else {
        alert('SPEAK UP, KID!');
    }
     deafGrandma(count)
}

deafGrandma();